CMakeLists.txt

```diff
@@ -1,22 +1,57 @@
 cmake_minimum_required(VERSION 3.26) # Required
```

Comment:
We could totally make bash scripts for the docker and cmake files to make it easier for users to install everything